[Doc] Added Inputs and Fields CSS Api documentation (#5346)
marmelab/react-admin: docs/Fields.md

`<BooleanField>` also accepts the [common field props](./Fields.md#common-field-props).

### CSS API

| Rule name  | Description                 |
| ---------- | --------------------------- |
| `root`     | Applied to the root element |

> **Tip**: `"RaBooleanField"`: Global css name for overriding all instances of the component using Material UI theme. See their [documentation](https://material-ui.com/customization/globals/#css).

Reviewer: ```suggestion
To override the style of all instances of `<BooleanField>` using the [material-ui style overrides](https://material-ui.com/customization/globals/#css), use the `RaBooleanField` key.
```

### Properties

> | Prop              | Required | Type                                         | Default | Description                                                                                                                                  |

Reviewer: Please refrain from reformatting tables like that - it's a never ending effort that has no real added value IMO.

Author: Same

| ----------------- | -------- | -------------------------------------------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------- |

> | `optionText`      | Optional | `string ` &#124; `Function` &#124; `Element` | 'name'  | Name of the field to use to display the matching choice, or function returning that field name, or a React element to render for that choice |

Reviewer: why did you change that? It renders fine in the doc site

Author: I belive this kind of formatting is necesary (correctme if I'm wrong), That's my understanding of this commit b625602e21d98aa9af3db8194816dafecb4743e6 made by @djhi regarding  `jekyll` warnings being thrown

### Properties

| Prop        | Required | Type                       | Default  | Description                                                                                                         |
| ----------- | -------- | -------------------------- | -------- | ------------------------------------------------------------------------------------------------------------------- |
| `reference` | Required | `string`                   | -        | The name of the resource for the referenced records, e.g. 'posts'                                                   |
| `children`  | Required | `Element`                  | -        | The Field element used to render the referenced record                                                              |
| `link`      | Optional | `string`                   | 'edit'   | Target of the link wrapping the rendered child                                                                      |

> | `sortBy`    | Optional | `string` &#124; `Function` | `source` | Name of the field to use for sorting when the user clicks on the column header. Set to `false` to disable the link. |

Reviewer: same

Author: Same

### Properties

| Prop         | Required | Type                       | Default                          | Description                                                                                                                                |
| ------------ | -------- | -------------------------- | -------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ |
| `reference`  | Required | `string`                   | -                                | The name of the resource for the referenced records, e.g. 'tags'                                                                           |
| `children`   | Required | `Element`                  | -                                | The Field element used to render the referenced records                                                                                    |

> | `sortBy`     | Optional | `string` &#124; `Function` | `source`                         | When used in a `List`, name of the field to use for sorting when the user clicks on the column header. Set to `false` to disable the link. |

Reviewer: same

Author: Same
